core/src/main/scala/com/raphtory/internals/serialisers/DynamicLoaderSerializer.scala

```diff
@@ -63,12 +63,13 @@ class DynamicLoaderSerializer(default: Serializer[DynamicLoader]) extends Serial
     // read how many classes are setup for Dynamic loading
     try {
       val n       = input.readInt()
+      val loader  = DynamicClassLoader(kryo.getClassLoader)
       val classes = (0 until n).map { _ =>
         val name   = input.readString()
         val length = input.readInt()
         val bytes  = input.readBytes(length)
-        DynamicClassLoader.injectClass(name, bytes, DynamicClassLoader(kryo.getClassLoader))
-      }.toSet
+        DynamicClassLoader.injectClass(name, bytes, loader)
+      }.toList
```

Comment:
I think we need a distinct here

Reply:
Adding distinct here is way too late. This is being handled when the DynamicLoader is resolved before sending it. I added a check in the write method such that it fails if we ever forget to do that.